### PR TITLE
Feat/input form validation

### LINKED ISSUE #6: Form Validation
## Value statement - a requirement, not a solution
As a user
I need to be forced to answer all questions within the correct formal criteria 
so that my bylaws will come out precisely

## Description (Text, Scribble, Wireframe, Design)
HelloBylawsValidation 2020-09-08

## Acceptance criteria
- [x] error message if an answer is missing completely 
- [x] no number inputs < 1 possible
- [x] number in question no. 23 must be min. 7 (due to min. number of founding members)
- [x] numbers in question no. 9 and 18 must be min. 14 
- [x] date (question no. 22) must be min. today's date
- [x] if question is answered after error message: error message will disappear with a click on the "Zusammenfassung anzeigen" button

## Tasks
- [x] create feature branch
- [x] validation and error message for text inputs
- [x] min. values for number inputs 
- [x] validation for date input (min date must be today's date)
- [x] validation and error message for number inputs
- [x] validation and error message for checkbox inputs
- [x] validation error message for radio inputs

## Size
medium

### PR DESCRIPTION
**Pull request by carolinwagner**
Please review. 

Preview deployment: https://capstone-project-git-feat-inputform-validation.carolinwagner.vercel.app/

Closes #6

### CODE REVIEW

File: src/QuestionsForm.js [hunk 1]
```diff
@@ -6,35 +6,51 @@ import TextInput from './TextInput'
...
 
 export default function QuestionsForm({ onClick }) {
-  const { register, handleSubmit } = useForm()
+  const { register, handleSubmit, watch, errors } = useForm({
+    reValidateMode: 'onSubmit',
+  })
 
-  const onSubmit = (data) => {
+  const onFormSubmit = (data) => {
     console.log(data)
```
Comment (ninasander) [inline src/QuestionsForm.js:18]: I think you forgot to remove this console log 😉

**Review by ninasander — APPROVED** [review]
Looks good to me! Just added a small comment :)

File: src/QuestionsForm.js (hunk 1 above)
Comment (carolinwagner) [reply src/QuestionsForm.js:18]: Haha thanks 😄

File: src/CheckboxInput.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
-import React, { forwardRef } from 'react'
+import React from 'react'
 import styled from 'styled-components/macro'
 
-const CheckboxInput = forwardRef(({ question }, ref) => {
+const CheckboxInput = ({ question, register, watch }) => {
   return (
     <div>
       {question.answerOptions.map((answerOption, innerIndex) => {
```
Comment (maxbischof) [inline src/CheckboxInput.js:7]: innerIndex? Its just the index isn't it?

File: src/CheckboxInput.js [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
-        const inputName = `${question?.questionText}[${innerIndex}]`
```
Comment (maxbischof) [inline src/CheckboxInput.js:8]: You could add proptypes if question is required.

File: src/CheckboxInput.js (hunk 2 above)
Comment (carolinwagner) [reply src/CheckboxInput.js:7]: Thanks, I fixed it :-)

File: src/CheckboxInput.js (hunk 3 above)
Comment (carolinwagner) [reply src/CheckboxInput.js:8]: Thanks, I consider it

**Pull request merged**